### PR TITLE
Van Code from Clem for Game Piece Selection

### PR DESCRIPTION
**Pull request by silverlightning926**
I *think* this should work. I've gone through and changed references from just a bare gamepiece and `getGamePiece()` to a Supplier<GamePiece>. AFAIK this should do what we want, and in less lines than our other solution. I have also added switch case for cubes where needed.

Nobody is allowed to talk about the spelling of the branch name.

### CODE REVIEW

**Comment by silverlightning926** [comment]
Hold up, forogt about autos. fixing now

File: src/main/java/frc/robot/commands/Presets/HighPresetSequence.java [hunk 1]
```diff
@@ -17,20 +19,25 @@
...
+
+    public HighPresetSequence(TelescopeSubsystem telescope, RotatorSubsystem rotator, CrocodileSubsystem crocodile,
+            DoubleSupplier intake, Supplier<GamePiece> gamePiece) {
         this.intake = intake;
-        addRequirements(rotator,telescope,crocodile);
-        
-        if(crocodile.getGamePiece() == GamePiece.CUBE){
+        addRequirements(rotator, telescope, crocodile);
+
+        if (gamePiece.get() == GamePiece.CUBE) {
```
Comment (mikemag) [inline src/main/java/frc/robot/commands/Presets/HighPresetSequence.java:29]: Calling your game piece lambda in the constructor!

**Review by mikemag — CHANGES_REQUESTED** [review]
Looks good, just missed one small place.

**Review by mikemag — APPROVED** [review]
Groovy

**Pull request merged**